float32 distance
float32 angleDiff
float32 Xdir
float32 Ydir
int8 edgePos
